pkg/globalnet/main.go

```diff
@@ -45,7 +47,16 @@ func main() {
 		klog.Fatalf("error building submariner clientset: %s", err.Error())
 	}
 
-	localCluster, err := submarinerClient.SubmarinerV1().Clusters(ipamSpec.Namespace).Get(ipamSpec.ClusterID, metav1.GetOptions{})
+	var localCluster *submarinerv1.Cluster
+	// During installation, sometimes creation of clusterCRD by submariner-gateway-pod would take few secs.
+	for i := 0; i < 5; i++ {
+		localCluster, err = submarinerClient.SubmarinerV1().Clusters(ipamSpec.Namespace).Get(ipamSpec.ClusterID, metav1.GetOptions{})
+		if err == nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
```

Comment:
Could 5 sec be fragile? Can we wait indefinitely - perhaps use a simple one-time informer with just an AddFunc and no queue?

Reply:
@tpantelis: As you know, this can happen ONLY during the very first installation of Submariner and what we have seen is that this does **not** happen frequently. TBH, I've never seen it with the KIND setup (both helm and operator deployments). IMHO we can have this 5 secs retry and if we continue to see the issue even after this change, we can revisit in future. I'm personally inclined with 5 seconds retry rather than indefinite as it will be an indication for the users too that globalnetPod went into an error - otherwise, they have to look at the logs to figure out.

Reply:
ok